smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java

```diff
@@ -185,6 +197,89 @@ public void verifySaveLayoutButton() {
+    /**
+     * This method tests whether the GraphMLTopologyProvider can work with categories - searching, collapsing and expanding
+     */
+    public void verifyCanFilterByCategory() throws IOException, InterruptedException {
+        topologyUIPage.selectTopologyProvider(() -> LABEL);
+        topologyUIPage.setSzl(0);
+        topologyUIPage.clearFocus();
+
+        // Search for the first category
+        topologyUIPage.search("Routers").selectItemThatContains("Routers");
+        Assert.assertNotNull(topologyUIPage.getVisibleVertices());
+        Assert.assertEquals(2, topologyUIPage.getVisibleVertices().size());
+        Assert.assertEquals("Cat_Rout", topologyUIPage.getVisibleVertices().get(0).getLabel());
+        Assert.assertEquals("Cat_Rout_Serv", topologyUIPage.getVisibleVertices().get(1).getLabel());
+
+        // Collapse and verify that collapsing works and that the category is visible while the vertex - not
+        topologyUIPage.getFocusedVertices().get(0).collapse();
+        Assert.assertEquals(1, topologyUIPage.getVisibleVertices().size());
+        Assert.assertEquals("Routers", topologyUIPage.getVisibleVertices().get(0).getLabel());
+
+        // Search for the second category
+        //TODO Theoretically it should display 2 vertices - one for each category. But it does not due to a bug (see issue NMS-9423)
+        topologyUIPage.search("Servers").selectItemThatContains("Servers");
+        Assert.assertNotNull(topologyUIPage.getVisibleVertices());
+        Assert.assertEquals(1, topologyUIPage.getVisibleVertices().size());
+        Assert.assertEquals("Routers", topologyUIPage.getVisibleVertices().get(0).getLabel());
+
+        // Expand and verify that vertices are visible again (and not duplicated)
+        for (TopologyIT.FocusedVertex vertex : topologyUIPage.getFocusedVertices()) {
+            vertex.expand();
+        }
+        Assert.assertEquals(2, topologyUIPage.getVisibleVertices().size());
+        Assert.assertEquals("Cat_Rout", topologyUIPage.getVisibleVertices().get(0).getLabel());
+        Assert.assertEquals("Cat_Rout_Serv", topologyUIPage.getVisibleVertices().get(1).getLabel());
+
+        // Collapse all and verify that vertex are not visible and that both categories are visible
+        for (TopologyIT.FocusedVertex vertex : topologyUIPage.getFocusedVertices()) {
+            vertex.collapse();
+        }
+        Assert.assertEquals(2, topologyUIPage.getVisibleVertices().size());
+        Assert.assertEquals("Routers", topologyUIPage.getVisibleVertices().get(0).getLabel());
+        Assert.assertEquals("Servers", topologyUIPage.getVisibleVertices().get(1).getLabel());
+    }
+
+    /**
+     * Creates dummy node with specified parameters
+     * @param foreignID Foreign ID
+     * @param label Node label
+     * @param categories List with node categories
+     */
+    private void createDummyNode(String foreignID, String label, List<String> categories) throws IOException, InterruptedException {
```

Comment:
This method should create the requisition for all nodes, not just one single node. I'd prefer it to just create a xml instead of using java StringBuilder.